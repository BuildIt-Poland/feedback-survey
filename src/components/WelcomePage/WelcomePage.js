import React, { useState, useEffect } from 'react';

import { spacingSmall, colorRed } from '../../styles/designTokens';
import Agreed from '../Agreed';
import LinkButton from '../Button/LinkButton';
import { surveyMatchParams } from '../../types';
import Page from '../Layout/Page';
import StyledHeading from '../Layout/Heading';
import Main from '../App/Main';

const WelcomePage = ({ match = { params: {} } }) => {
  const [surveyId, setSurveyID] = useState('');
  const [employeeName, setEmployeeName] = useState('');
  const [path, setPath] = useState('');

  useEffect(() => {
    setSurveyID(match.params.surveyId ? match.params.surveyId : '');
    setEmployeeName(match.params.employeeName ? match.params.employeeName : '');
    setPath(surveyId && employeeName ? `/survey/${surveyId}/${employeeName}` : '/survey');
  });

  return (
    <Main bgImage="/image/TextBalloons_illustration.png">
      <Page>
        <StyledHeading level="1" margin={{ bottom: 'none', top: 'none' }} color={colorRed}>
          Performance Snapshot on RSR Consultant:
        </StyledHeading>
        <StyledHeading level="2" margin={{ top: spacingSmall, bottom: spacingSmall }} color={colorRed}>
          {employeeName}
        </StyledHeading>
        <Agreed />
        <LinkButton linkTo={path} label="start" />
      </Page>
    </Main>
  );
};

WelcomePage.propTypes = {
  match: surveyMatchParams
};

export default WelcomePage;
