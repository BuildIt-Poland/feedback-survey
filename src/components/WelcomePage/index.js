import { withRouter } from 'react-router-dom';

import WelcomePage from './WelcomePage';

export default withRouter(WelcomePage);
