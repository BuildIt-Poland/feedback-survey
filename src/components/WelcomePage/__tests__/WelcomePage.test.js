import React from 'react';
import { create } from 'react-test-renderer';

import WelcomePage from '../WelcomePage';

jest.mock('react-router-dom', () => ({
  Link: 'Link'
}));

describe('COMPONENT - WelcomePage', () => {
  it('renders correctly with params', () => {
    const match = {
      params: {
        surveyId: '12345',
        employeeName: 'John'
      }
    };
    const component = create(<WelcomePage match={match} />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly without params', () => {
    const component = create(<WelcomePage />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
