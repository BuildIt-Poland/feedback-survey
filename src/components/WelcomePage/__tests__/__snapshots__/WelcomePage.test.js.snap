// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - WelcomePage renders correctly with params 1`] = `
<main
  className="sc-EHOje kfEkWq"
>
  <section
    className="sc-bxivhb dFHFAh"
  >
    <h1
      className="sc-ifAKCX fqciGW StyledHeading-sc-1rdh4aw-0 cnPzet"
    >
      Performance Snapshot on Consulting Talent - Consultant:
    </h1>
    <h2
      className="sc-ifAKCX fqciGW StyledHeading-sc-1rdh4aw-0 jgvbu"
    >
      
    </h2>
    <section
      className="sc-bwzfXH iPoewc"
    >
      <p
        className="sc-bdVaJa gxYkJN"
      >
        The feedback you provide will be used for individual development and career pathway and forms part of RSR Reward and Recognition scheme.
      </p>
      <p
        className="sc-bdVaJa gxYkJN"
      >
        Feedback will also be used to enhance and evolve individual contribution to your organisation.
      </p>
      <p
        className="sc-bdVaJa gxYkJN"
      >
        We therefore ask you to provide feedback based on factual evidence only and we may ask you for further information in some individual performance feedback.
      </p>
    </section>
    <Link
      to=""
    >
      <button
        className="sc-htpNat dBprKw StyledButton-sc-323bzc-0 bmExQD"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        size="large"
        type="button"
      >
        start
      </button>
    </Link>
  </section>
</main>
`;

exports[`COMPONENT - WelcomePage renders correctly without params 1`] = `
<main
  className="sc-EHOje kfEkWq"
>
  <section
    className="sc-bxivhb dFHFAh"
  >
    <h1
      className="sc-ifAKCX fqciGW StyledHeading-sc-1rdh4aw-0 cnPzet"
    >
      Performance Snapshot on Consulting Talent - Consultant:
    </h1>
    <h2
      className="sc-ifAKCX fqciGW StyledHeading-sc-1rdh4aw-0 jgvbu"
    >
      
    </h2>
    <section
      className="sc-bwzfXH iPoewc"
    >
      <p
        className="sc-bdVaJa gxYkJN"
      >
        The feedback you provide will be used for individual development and career pathway and forms part of RSR Reward and Recognition scheme.
      </p>
      <p
        className="sc-bdVaJa gxYkJN"
      >
        Feedback will also be used to enhance and evolve individual contribution to your organisation.
      </p>
      <p
        className="sc-bdVaJa gxYkJN"
      >
        We therefore ask you to provide feedback based on factual evidence only and we may ask you for further information in some individual performance feedback.
      </p>
    </section>
    <Link
      to="/survey"
    >
      <button
        className="sc-htpNat dBprKw StyledButton-sc-323bzc-0 bmExQD"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        size="large"
        type="button"
      >
        start
      </button>
    </Link>
  </section>
</main>
`;
